hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -374,6 +395,191 @@ def finalize(self, settings, item):
+
+        # Add a combobox to edit the publish mode
+        publish_mode_label = QtGui.QLabel("Publish Mode:")
+        publish_mode_combobox = QtGui.QComboBox(widget)
+        publish_mode_combobox.setAccessibleName("Publish mode selection dropdown")
+        publish_mode_combobox.addItems(self.PUBLISH_MODE_OPTIONS)
+        publish_mode_widget = QtGui.QWidget(widget)
+        publish_mode_widget.setSizePolicy(
+            QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Preferred
+        )
+        publish_mode_layout = QtGui.QHBoxLayout()
+        publish_mode_layout.setContentsMargins(0, 0, 0, 0)
+        publish_mode_layout.addWidget(publish_mode_label)
+        publish_mode_layout.addWidget(publish_mode_combobox)
+        publish_mode_layout.addStretch()
+        publish_mode_widget.setLayout(publish_mode_layout)
+
+        # Add all the minor widgets to the main widget
+        widget_layout.addWidget(publish_mode_widget)
+        widget_layout.addWidget(description_groupbox)
+        widget.setLayout(widget_layout)
+
+        # Set the widget property to store the combobox to access in get_ui_settings and set_ui_settings
+        widget.setProperty("publish_mode_combobox", publish_mode_combobox)
+
+        return widget
+
+    def get_ui_settings(self, widget, items=None):
+        """
+        This method is required to be defined in order for the custom UI to show up in the app.
+
+        Invoked by the Publisher when the selection changes. This method gathers the settings
+        on the previously selected task, so that they can be later used to repopulate the
+        custom UI if the task gets selected again. They will also be passed to the accept, validate,
+        publish and finalize methods, so that the settings can be used to drive the publish process.
+
+        The widget argument is the widget that was previously created by
+        `create_settings_widget`.
+
+        The method returns a dictionary, where the key is the name of a
+        setting that should be updated and the value is the new value of that
+        setting. Note that it is up to you how you want to store the UI's state as
+        settings and you don't have to necessarily to return all the values from
+        the UI. This is to allow the publisher to update a subset of settings
+        when multiple tasks have been selected.
+
+        Example::
+
+            {
+                 "setting_a": "/path/to/a/file"
+            }
+
+        :param widget: The widget that was created by `create_settings_widget`
+        """
+
+        ui_settings = {}
+
+        # Get the Publish Mode settings value from the UI combobox
+        publish_mode_combobox = widget.property("publish_mode_combobox")
+        if publish_mode_combobox:
+            mode_index = publish_mode_combobox.currentIndex()
+            if 0 <= mode_index < len(self.PUBLISH_MODE_OPTIONS):
+                self.PUBLISH_MODE_OPTIONS[mode_index]
```

Comment:
Is this line useful?

Reply:
Oops, good catch. This was not intended to be here